sklearn/naive_bayes.py

```diff
@@ -1528,3 +1541,600 @@ def _joint_log_likelihood(self, X):
+    >>> clf.fit(X, y)
+    ColumnwiseNB(nb_estimators=[('mnb1', MultinomialNB(), [0, 1]),
+                            ('mnb2', MultinomialNB(), [3, 4]),
+                            ('gnb1', GaussianNB(), [5])])
+    >>> print(clf.predict(X))
+    [0 0 1 0 2 2]
+    """
+
+    _required_parameters = ["nb_estimators"]
+
+    _parameter_constraints = {
+        "nb_estimators": "no_validation",
+        "priors": ["array-like", str, None],
+        "n_jobs": [Integral, None],
+        "verbose": ["verbose"],
+    }
+
+    def _log_message(self, name, idx, total):
+        if not self.verbose:
+            return None
+        return "(%d of %d) Processing %s" % (idx, total, name)
+
+    def __init__(self, nb_estimators, *, priors=None, n_jobs=None, verbose=False):
+        self.nb_estimators = nb_estimators
+        self.priors = priors
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def _check_X(self, X):
+        """Validate X, used only in predict* methods."""
+        # The meta-estimator checks for feature names only. Other checks
+        # and conversion to numpy array are performed by subestimators.
+        # It is important that X is not modified by the meta-estimator, and X's
+        # columns are passed to an estimator as they are. Note that estimators
+        # may modify (a copy of) X. E.g., BernoulliNB._check_X binarises the
+        # input.
+        self._check_feature_names(X, reset=False)
+        return X
+
+    def _joint_log_likelihood(self, X):
+        """Calculate the meta-estimator's joint log-probability ``log P(x,y)``."""
+        estimators = self._iter(fitted=True, replace_strings=True)
+        all_jlls = Parallel(n_jobs=self.n_jobs)(
+            delayed(_jll_one)(estimator=nb_estimator, X=_safe_indexing(X, cols, axis=1))
+            for (_, nb_estimator, cols) in estimators
+        )
+        n_estimators = len(all_jlls)
+        log_prior = np.log(self.class_prior_)
+        return np.sum(all_jlls, axis=0) - (n_estimators - 1) * log_prior
+
+    def _validate_estimators(self, check_partial=False):
+        # Check if estimators have fit/partial_fit and joint log prob methods
+        # Validate estimator names via _BaseComposition._validate_names(self, names)
+        if not self.nb_estimators:
+            raise ValueError(
+                "A list of naive Bayes estimators must be provided "
+                "in the form [(name, nb_estimator, columns), ... ]."
+            )
+        names, estimators, _ = zip(*self.nb_estimators)
+        for e in estimators:
+            if (not check_partial) and (
+                not (hasattr(e, "fit") and hasattr(e, "predict_joint_log_proba"))
+            ):
+                raise TypeError(
+                    "Estimators must be naive Bayes estimators implementing "
+                    "`fit` and `predict_joint_log_proba` methods."
+                )
+            if check_partial and (
```

Comment:
We will not need this one because we delegate to `available_if` decorator.

Reply:
I changed this slightly. I am still permitting the situation in which there is `partial_fit`, but no `fit`.